tests/test_external_search.py

```diff
@@ -991,48 +991,168 @@ class TestSearchOrder(EndToEndSearchTest):
     def populate_works(self):
         _work = self.default_work
 
-        # Create two works -- this part is straightforward.
+        # We're going to create three works:
+        # a: "Moby Dick"
+        # b: "Moby Duck"
+        # c: "[untitled]"
+        #
+        # The metadata of these books will be set up to generate
+        # intuitive orders under most of the ordering scenarios.
+        #
+        # The most complex ordering scenario is ORDER_LAST_UPDATE,
+        # which orders books differently depending on the modification
+        # date of the Work, the date a LicensePool for the work was
+        # first seen in a collection associated with the filter, and
+        # the date the work was first seen on a custom list associated
+        # with the filter.
+        #
+        # The modification dates of the works will be set in the order
+        # of their creation.
+        #
+        # We're going to put all three works in two different
+        # collections with different dates. All three works will be
+        # added to two different custom lists, and works a and c will
+        # be added to a third custom list.
+        #
+        # The dates associated with the "collection add" and "list add"
+        # events will be set up to create the following orderings:
+        #
+        # a, b, c - when no collections or custom lists are associated with
+        #           the Filter.
+        # a, c, b - when collection 1 is associated with the Filter.
+        # b, a, c - when collections 1 and 2 are associated with the Filter.
+        # b, c, a - when custom list 1 is associated with the Filter.
+        # c, a, b - when collection 1 and custom list 2 are associated with
+        #           the Filter.
+        # c, a - when two sets of custom list restrictions [1], [3]
+        #        are associated with the filter.
```

Comment:
This is pretty thorough. The only thing I can think of that's not covered is when the works are associated with collections or lists, but the works' last update times are more recent and control the order.